articles/embedded.md

```diff
@@ -50,48 +50,44 @@ To determine how feasible that is, and to come up with the actual requirements a
+*   [**EU project OFERA**](http://ofera.eu/): The EU project OFERA (Open Framework for Embedded Robot Applications) aims at a ROS 2-compatible stack for MCUs in the range of STM32F4 or STM32L1, i.e. with possibly less than 100kB RAM. The project partners currently investigate of using the ROS 2 rmw, rcl and rclcpp layers as-is on the micro-XRCE-DDS implementation of the upcoming XRCE-DDS standard. In parallel, a more modular approach in the style of rosserial is investigated. In the project's use-cases, NuttX is considered as primary choice for the RTOS. Beyond the project page, additional details of the project results can be found at [micro-ros.com](http://micro-ros.com).
+
+*   [**ROS 2 library for OpenCR by ROBOTIS**](https://github.com/ROBOTIS-GIT/OpenCR/tree/feature-ros2-micrortps/arduino/opencr_arduino/opencr/libraries/ROS2): Tailored and optimized implementation of the ROS 2 publish/subscribe and clock API for the micro-XRCE (formerly micro-RTPS) implementation of the upcoming XRCE-DDS middleware standard running on an STM32F7.
 
-*   [ROS 2 library for OpenCR by ROBOTIS](https://github.com/ROBOTIS-GIT/OpenCR/tree/feature-ROS2-micrortps/arduino/opencr_arduino/opencr/libraries/ROS2)
+*   [**freeRTPS**](https://github.com/ros2/freertps): A free, portable, minimalist implementation of the RTPS protocol for microcontrollers such as the STM32F7 developed at the OSRF. FreeRTPS shall allow to run ROS 2 with standard DDS as-is on stronger MCUs.
 
+*   [**ros2_embedded_nuttx**](https://github.com/ros2/ros2_embedded_nuttx): Early port (in 2014) of ROS 2 alpha for the STM32F4Discovery board and the STM3240G eval board running the RTOS NuttX developed by Víctor Mayoral Vilches and Esteve Fernandez at the OSRF.
 
 ### ROS1-based approaches
 
-*   [rosserial](http://wiki.ros.org/rosserial) -- well-known and widely used in the ROS community ...
+*   [**rosserial**](http://wiki.ros.org/rosserial): Well-known and widely used in the ROS community.
 
-*   [mROS](https://github.com/tlk-emb/mROS/) -- a new work on bringing ROS1 concepts (including nodes and the ROS1 middleware) on stronger MCUs, cf. also
+*   [**mROS**](https://github.com/tlk-emb/mROS/): A new work on bringing ROS1 concepts (including nodes and the ROS1 middleware) on stronger MCUs, cf.
     *Hideki Takase, Tomoya Mori, Kazuyoshi Takagi and Naofumi Takagi: 'Work-in-Progress: Design Concept of a Lightweight Runtime Environment for Robot Software Components onto Embedded Devices' in Proc. of ESWEEK, Torino, Italy, September 2018.*
-*   [freeRTPS](TODO): TODO
-
-*   [ros2_embedded_nuttx](TODO): TODO
 
 ## First Design Ideas
 
 * This is our proposed architecture:
 
 ![micro-ROS](/img/embedded/micro-ROS_proposed_architecture.png)
```

Comment:
I propose to replace this architecture diagram with the more abstract diagram by Borja. The current diagram is rather fine-grained given the abstract discussion above where as Borjas diagram could be a good starting point to (1.) discuss the desired feature set for an embedded ROS 2 stack and (2.) the implications and dependencies with regard to the lower layers.

Reply:
We already have it exposed to the community at micro-ros.com. During the course of the discussion on this article, we probably reach the point where we can put the detailed diagram on the table. Yet, for the start, I would prefer a more abstract view focussed on the API features.

Reply:
@BorjaOuterelo I just added your diagram and a link to the other diagram on micro-ros.com. I propose to replace 'rclc' with 'rclc/rclcpp' and to fix the name of the middleware. Yet, I did not find the sources in our gitlab repo. Can you point me to the corresponding file or change it yourself, please?